scipy/stats/tests/test_sampling.py

```diff
@@ -582,5 +598,226 @@ def test_bad_urn_factor(self):
     def test_bad_args(self):
         msg = (r"`domain` must be provided when the "
                r"probability vector is not available.")
+
+        class dist:
+            def pmf(self, x):
+                return x
+
+        with pytest.raises(ValueError, match=msg):
+            DiscreteAliasUrn(dist)
+
+
+class TestNumericalInversePolynomial:
+    # Simple Custom Distribution
+    class dist0:
+        def pdf(self, x):
+            return 3/4 * (1-x*x)
+
+        def cdf(self, x):
+            return 3/4 * (x - x**3/3 + 2/3)
+
+        def support(self):
+            return -1, 1
+
+    # Standard Normal Distribution
+    class dist1:
+        def pdf(self, x):
+            return stats.norm._pdf(x / 0.1)
+
+        def cdf(self, x):
+            return stats.norm._cdf(x / 0.1)
+
+    # Sin 2 distribution
+    #          /  0.05 + 0.45*(1 +sin(2 Pi x))  if |x| <= 1
+    #  f(x) = <
+    #          \  0        otherwise
+    # Taken from UNU.RAN test suite (from file t_pinv.c)
+    class dist2:
+        def pdf(self, x):
+            return 0.05 + 0.45 * (1 + np.sin(2*np.pi*x))
+
+        def cdf(self, x):
+            return (0.05*(x + 1) +
+                    0.9*(1. + 2.*np.pi*(1 + x) - np.cos(2.*np.pi*x)) /
+                    (4.*np.pi))
+
+        def support(self):
+            return -1, 1
+
+    # Sin 10 distribution
+    #          /  0.05 + 0.45*(1 +sin(2 Pi x))  if |x| <= 5
+    #  f(x) = <
+    #          \  0        otherwise
+    # Taken from UNU.RAN test suite (from file t_pinv.c)
+    class dist3:
+        def pdf(self, x):
+            return 0.2 * (0.05 + 0.45 * (1 + np.sin(2*np.pi*x)))
+
+        def cdf(self, x):
+            return x/10. + 0.5 + 0.09/(2*np.pi) * (np.cos(10*np.pi) -
+                                                   np.cos(2*np.pi*x))
+
+        def support(self):
+            return -5, 5
+
+    dists = [dist0(), dist1(), dist2(), dist3()]
+
+    # exact mean and variance of the distributions in the list dists
+    mv0 = [0., 4./15.]
+    mv1 = [0., 0.01]
+    mv2 = [-0.45/np.pi, 2/3*0.5 - 0.45**2/np.pi**2]
+    mv3 = [-0.45/np.pi, 0.2 * 250/3 * 0.5 - 0.45**2/np.pi**2]
+    mvs = [mv0, mv1, mv2, mv3]
+
+    @pytest.mark.parametrize("dist, mv_ex",
+                             zip(dists, mvs))
+    def test_basic(self, dist, mv_ex):
+        rng = NumericalInversePolynomial(dist, random_state=42)
+        check_cont_samples(rng, dist, mv_ex)
+
+    very_slow_dists = ['studentized_range', 'trapezoid', 'triang', 'vonmises',
+                       'levy_stable', 'kappa4', 'ksone', 'kstwo', 'levy_l',
+                       'gausshyper', 'beta', 'anglit']
+    # for some reason, UNU.RAN segmentation faults for the uniform and
+    # arcsine distribution!
+    fatal_fail_dists = ['uniform', 'arcsine']
```

Comment:
arcsince is unbounded, we could add it to a list `fails_undbounded_pdf`

not sure why uniform does not work, maybe because it is constant

Reply:
`beta` can also be added to unbounded (default b < 1)

Reply:
Done.